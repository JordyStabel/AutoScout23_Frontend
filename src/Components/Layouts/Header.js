import React from "react";
import {AppBar, Toolbar, Typography} from "@material-ui/core";
import CreateDialog from "../Content/Create";
import FormControlLabel from "@material-ui/core/FormControlLabel";
import Switch from "@material-ui/core/Switch";
//import {toggleTheme} from "../../index";

let isChecked = false;

function themeSwitch() {
    //toggleTheme();
    isChecked = !isChecked;
}

export default ({allergies, categories, onDishCreate, makes, toggleShowAll, showAll, onRequestBill, order}) => (
    <AppBar position="static">
        <Toolbar>
            <Typography variant="title" color="inherit" style={{flex: 1, marginLeft: 20}}>
                AutoScout23 - Buy & Sell Cars
            </Typography>
            <FormControlLabel
                control={
                    <Switch
                        color="secondary"
                        onChange={themeSwitch}
                    />
                }
                label={isChecked ? "Light-mode" : "Dark-mode"}
            />
            {/*<FormControlLabel*/}
                {/*control={*/}
                    {/*<Switch*/}
                        {/*color="secondary"*/}
                        {/*onChange={toggleShowAll}*/}
                    {/*/>*/}
                {/*}*/}
                {/*label={showAll ? "Show selected dishes" : "Show all dishes"}*/}
            {/*/>*/}
            <CreateDialog allergies={allergies} categories={categories} makes={makes} onCreate={onDishCreate}/>
        </Toolbar>
    </AppBar>
);
