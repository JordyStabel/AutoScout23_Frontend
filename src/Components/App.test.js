import { add } from "./App";
import React from "react";
import { shallow } from "enzyme";
import Header from "./Layouts/Header";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({adapter: new Adapter()});

describe('Header', () => {
    test('Rendering header', () =>{
        const wrapper = shallow(
            <Header/>
        );

        expect(wrapper).toMatchSnapshot();
    });
});

test('Adding numbers', () => {
    expect(add(4,6)).toBe(10);
});

test("Fake Test", () => {
    expect(true).toBeTruthy();
});

test("Fake Test II", () => {
    expect(false).toBeFalsy();
});