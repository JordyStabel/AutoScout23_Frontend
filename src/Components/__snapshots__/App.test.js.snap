// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Rendering a simple component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <_default />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <WithStyles(Toolbar)>
        <WithStyles(Typography)
          color="inherit"
          style={
            Object {
              "flex": 1,
              "marginLeft": 20,
            }
          }
          variant="title"
        >
          AutoScout23 - Buy & Sell Cars
        </WithStyles(Typography)>
        <WithStyles(FormControlLabel)
          control={
            <WithStyles(Switch)
              color="secondary"
              onChange={[Function]}
            />
          }
          label="Dark-mode"
        />
        <_default />
      </WithStyles(Toolbar)>,
      "position": "static",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WithStyles(Typography)
            color="inherit"
            style={
              Object {
                "flex": 1,
                "marginLeft": 20,
              }
            }
            variant="title"
          >
            AutoScout23 - Buy & Sell Cars
          </WithStyles(Typography)>,
          <WithStyles(FormControlLabel)
            control={
              <WithStyles(Switch)
                color="secondary"
                onChange={[Function]}
              />
            }
            label="Dark-mode"
          />,
          <_default />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "AutoScout23 - Buy & Sell Cars",
            "color": "inherit",
            "style": Object {
              "flex": 1,
              "marginLeft": 20,
            },
            "variant": "title",
          },
          "ref": null,
          "rendered": "AutoScout23 - Buy & Sell Cars",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "control": <WithStyles(Switch)
              color="secondary"
              onChange={[Function]}
            />,
            "label": "Dark-mode",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "allergies": undefined,
            "categories": undefined,
            "makes": undefined,
            "onCreate": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <WithStyles(Toolbar)>
          <WithStyles(Typography)
            color="inherit"
            style={
              Object {
                "flex": 1,
                "marginLeft": 20,
              }
            }
            variant="title"
          >
            AutoScout23 - Buy & Sell Cars
          </WithStyles(Typography)>
          <WithStyles(FormControlLabel)
            control={
              <WithStyles(Switch)
                color="secondary"
                onChange={[Function]}
              />
            }
            label="Dark-mode"
          />
          <_default />
        </WithStyles(Toolbar)>,
        "position": "static",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(Typography)
              color="inherit"
              style={
                Object {
                  "flex": 1,
                  "marginLeft": 20,
                }
              }
              variant="title"
            >
              AutoScout23 - Buy & Sell Cars
            </WithStyles(Typography)>,
            <WithStyles(FormControlLabel)
              control={
                <WithStyles(Switch)
                  color="secondary"
                  onChange={[Function]}
                />
              }
              label="Dark-mode"
            />,
            <_default />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "AutoScout23 - Buy & Sell Cars",
              "color": "inherit",
              "style": Object {
                "flex": 1,
                "marginLeft": 20,
              },
              "variant": "title",
            },
            "ref": null,
            "rendered": "AutoScout23 - Buy & Sell Cars",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "control": <WithStyles(Switch)
                color="secondary"
                onChange={[Function]}
              />,
              "label": "Dark-mode",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "allergies": undefined,
              "categories": undefined,
              "makes": undefined,
              "onCreate": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
